go/vt/vtgate/planbuilder/testdata/filter_cases.txt

```diff
@@ -733,7 +733,46 @@ Gen4 plan same as above
     ]
   }
 }
-Gen4 plan same as above
+{
+  "QueryType": "SELECT",
+  "Original": "select user_extra.id from user join user_extra on user.col = user_extra.col where user.id = 5",
+  "Instructions": {
+    "OperatorType": "Join",
+    "Variant": "HashJoin",
+    "ComparisonType": "INT16",
+    "JoinColumnIndexes": "2",
+    "Predicate": "`user`.col = user_extra.col",
+    "TableName": "`user`_user_extra",
+    "Inputs": [
+      {
+        "OperatorType": "Route",
+        "Variant": "SelectEqualUnique",
```

Comment:
this looks weird. why do we want a SelectEqualUnique on the LHS and a Scatter on the rhs?